python/lsst/ctrl/mpexec/cmdLineFwk.py

```diff
@@ -353,11 +353,17 @@ def makeWriteButler(cls, args: argparse.Namespace) -> Butler:
             chainDefinition = list(self.output.chain if self.output.exists else self.inputs)
             if args.replace_run:
                 replaced, _ = chainDefinition.pop(0)
-                if args.prune_replaced:
-                    # TODO: DM-23671: need a butler API for pruning an
-                    # entire RUN collection, then apply it to 'replaced'
-                    # here.
-                    raise NotImplementedError("Support for --prune-replaced is not yet implemented.")
+                if args.prune_replaced == "unstore":
+                    # Remove datasets from datastore
+                    with butler.transaction():
+                        refs = butler.registry.queryDatasets(..., collections=replaced, deduplicate=True)
+                        butler.pruneDatasets(refs, unstore=True, run=replaced, disassociate=False)
+                elif args.prune_replaced == "purge":
+                    # Erase entire collection and all datasets, need to remove
+                    # collection from its chain collection first.
+                    with butler.transaction():
+                        butler.registry.setCollectionChain(self.output.name, chainDefinition)
+                        butler.pruneCollection(replaced, purge=True, unstore=True)
```

Comment:
can you have an else that bails out here? I know that config is supposed to restrict the options but I still feel better if we check.

Reply:
I will add handling for all possible options.